src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -0,0 +1,850 @@
+            case BUILD_TYPE_DEPLOYMENT:
+            case BUILD_TYPE_RETRY:
+                Console::info('Creating build for deployment: ' . $deployment->getId());
+                $github = new GitHub($cache);
+                $this->buildDeployment($deviceForFunctions, $queueForFunctions, $queueForEvents, $queueForUsage, $dbForConsole, $dbForProject, $github, $project, $resource, $deployment, $template, $log);
+                break;
+
+            default:
+                throw new \Exception('Invalid build type');
+        }
+    }
+
+    /**
+     * @param Device $deviceForFunctions
+     * @param Func $queueForFunctions
+     * @param Event $queueForEvents
+     * @param Usage $queueForUsage
+     * @param Database $dbForConsole
+     * @param Database $dbForProject
+     * @param GitHub $github
+     * @param Document $project
+     * @param Document $resource
+     * @param Document $deployment
+     * @param Document $template
+     * @param Log $log
+     * @return void
+     * @throws \Utopia\Database\Exception
+     * @throws Exception
+     */
+    protected function buildDeployment(Device $deviceForFunctions, Func $queueForFunctions, Event $queueForEvents, Usage $queueForUsage, Database $dbForConsole, Database $dbForProject, GitHub $github, Document $project, Document $resource, Document $deployment, Document $template, Log $log): void
+    {
+        // todo: refactor
+        $isFunction = $resource->getCollection() === 'functions';
+        $isSite = $resource->getCollection() === 'sites';
+        $foreignKey = $isFunction ? 'functionId' : 'siteId';
+
+        $executor = new Executor(System::getEnv('_APP_EXECUTOR_HOST'));
+
+        $log->addTag($foreignKey, $resource->getId());
+
+        $resource = $dbForProject->getDocument($resource->getCollection(), $resource->getId());
+        if ($resource->isEmpty()) {
+            throw new \Exception('Function not found', 404);
+        }
+
+        $log->addTag('deploymentId', $deployment->getId());
+
+        $deployment = $dbForProject->getDocument('deployments', $deployment->getId());
+        if ($deployment->isEmpty()) {
+            throw new \Exception('Deployment not found', 404);
+        }
+
+        if (empty($deployment->getAttribute('entrypoint', ''))) {
+            throw new \Exception('Entrypoint for your Appwrite Function is missing. Please specify it when making deployment or update the entrypoint under your function\'s "Settings" > "Configuration" > "Entrypoint".', 500);
+        }
+
+        $version = $resource->getAttribute('version', 'v2');
+        $spec = Config::getParam('runtime-specifications')[$resource->getAttribute('specifications', APP_FUNCTION_SPECIFICATION_DEFAULT)];
+        $runtimes = Config::getParam($version === 'v2' ? 'runtimes-v2' : 'runtimes', []);
+        // todo: fix for sites using frameworks
+        $key =  $resource->getAttribute('runtime');
+        $runtime = $runtimes[$key] ?? null;
+        if (\is_null($runtime)) {
```

Comment:
Sites do not have a `runtime`. They have a `framework`. We need to find `buildRuntime` and `serveRuntime` using a config file.